### PR TITLE
chore: automate adding heartbeat for jobs

### PR DESCRIPTION
**Pull request by Hamza442**

### CODE REVIEW

File: dataeng/resources/opsgenie-enable-heartbeat.sh [hunk 1]
```diff
@@ -25,7 +25,23 @@ if [ -n "$OPSGENIE_HEARTBEAT_NAME" ] && \
     HEARTBEAT_API_PING_URL="$HEARTBEAT_API_URL/ping"
     ALERT_MESSAGE="Heartbeat [$OPSGENIE_HEARTBEAT_NAME] expired - job is likely stuck."
     JSON_REQ_DATA="{ \"interval\": \"$OPSGENIE_HEARTBEAT_DURATION_NUM\", \"intervalUnit\": \"$OPSGENIE_HEARTBEAT_DURATION_UNIT\", \"alertMessage\": \"$ALERT_MESSAGE\", \"enabled\": true }"
```
Comment (pwnage101) [inline dataeng/resources/opsgenie-enable-heartbeat.sh:27]: Nit: IMO, this `JSON_REQ_DATA` variable definition should be next to the next section after the heartbeat is created.

File: dataeng/resources/opsgenie-enable-heartbeat.sh [hunk 2]
```diff
@@ -25,7 +25,23 @@ if [ -n "$OPSGENIE_HEARTBEAT_NAME" ] && \
...
-
+    
+    # Make API request to get existing heartbeat
+    # If the heartbeat already exist, then GET_EXISTING_HEARTBEAT will set to heartbeat name "GET_EXISTING_HEARTBEAT=heatbeat_name"
+    # If heartbeat don't exist GET_EXISTING_HEARTBEAT is set to empty
+    GET_EXISTING_HEARTBEAT=$(curl -X GET $HEARTBEAT_API_URL --header "$AUTH_HEADER" | grep -o $OPSGENIE_HEARTBEAT_NAME | sort -u)
+     # Add the heartbeat if heartbeat doesn't exist
+    if [ -z "$GET_EXISTING_HEARTBEAT" ]; then
+        JSON_REQ_HEARTBEAT_DATA="{\"name\":\"$OPSGENIE_HEARTBEAT_NAME\"
```
Comment (pwnage101) [inline dataeng/resources/opsgenie-enable-heartbeat.sh:35]: Can you rename this to `JSON_REQ_DATA_CREATE_HEARTBEAT`

File: dataeng/resources/opsgenie-enable-heartbeat.sh [hunk 3]
```diff
@@ -25,7 +25,23 @@ if [ -n "$OPSGENIE_HEARTBEAT_NAME" ] && \
     HEARTBEAT_API_PING_URL="$HEARTBEAT_API_URL/ping"
     ALERT_MESSAGE="Heartbeat [$OPSGENIE_HEARTBEAT_NAME] expired - job is likely stuck."
     JSON_REQ_DATA="{ \"interval\": \"$OPSGENIE_HEARTBEAT_DURATION_NUM\", \"intervalUnit\": \"$OPSGENIE_HEARTBEAT_DURATION_UNIT\", \"alertMessage\": \"$ALERT_MESSAGE\", \"enabled\": true }"
-
+    
+    # Make API request to get existing heartbeat
+    # If the heartbeat already exist, then GET_EXISTING_HEARTBEAT will set to heartbeat name "GET_EXISTING_HEARTBEAT=heatbeat_name"
+    # If heartbeat don't exist GET_EXISTING_HEARTBEAT is set to empty
+    GET_EXISTING_HEARTBEAT=$(curl -X GET $HEARTBEAT_API_URL --header "$AUTH_HEADER" | grep -o $OPSGENIE_HEARTBEAT_NAME | sort -u)
```
Comment (pwnage101) [inline dataeng/resources/opsgenie-enable-heartbeat.sh:32]: is `| sort -u` essential for functionality, or can it be removed?

Comment (Hamza442) [reply dataeng/resources/opsgenie-enable-heartbeat.sh:32]: The json response has heartbeat name more than one time, if I do grep -o heartbeat_name it returns duplicates so it **sort -u** de-depulicate and returns heartbeat name only once so I think its better to keep it there

**Review by mzakaria1 — APPROVED** [review]

File: dataeng/resources/opsgenie-enable-heartbeat.sh [hunk 4]
```diff
@@ -24,8 +24,25 @@ if [ -n "$OPSGENIE_HEARTBEAT_NAME" ] && \
     HEARTBEAT_API_URL="$OPSGENIE_HEARTBEAT_API_URL/$OPSGENIE_HEARTBEAT_NAME"
     HEARTBEAT_API_PING_URL="$HEARTBEAT_API_URL/ping"
     ALERT_MESSAGE="Heartbeat [$OPSGENIE_HEARTBEAT_NAME] expired - job is likely stuck."
+    
+    # Make API request to get existing heartbeat
+    # If the heartbeat already exist, then GET_EXISTING_HEARTBEAT will set to heartbeat name "GET_EXISTING_HEARTBEAT=heatbeat_name"
```
Comment (HassanJaveed84) [inline dataeng/resources/opsgenie-enable-heartbeat.sh:29]: nit: exist -> exists

File: dataeng/resources/opsgenie-enable-heartbeat.sh [hunk 5]
```diff
@@ -24,8 +24,25 @@ if [ -n "$OPSGENIE_HEARTBEAT_NAME" ] && \
...
+    # If heartbeat don't exist GET_EXISTING_HEARTBEAT is set to empty
```
Comment (HassanJaveed84) [inline dataeng/resources/opsgenie-enable-heartbeat.sh:30]: nit: don't -> does not

**Review by HassanJaveed84 — APPROVED** [review]
Just a couple of nits, otherwise LGTM.

**Pull request merged**